### PR TITLE
feat(worker): manage default workers config through env variables

### PR DESCRIPTION
**Pull request by p-fernandez**
### What change does this PR introduce?

Allow control the default values for the workers through environment variables.

### Why was this change needed?

We need to study which are the right values for concurrency in our workers. 
Also it is a first step to be able to create a generic worker that could be instantiated with certain configuration by IaC through environment variables to have dedicated fleets.

### CODE REVIEW

File: apps/api/src/config/env-validator.ts [hunk 1]
```diff
@@ -73,6 +72,12 @@ const validators: { [K in keyof any]: ValidatorSpec<any[K]> } = {
   LAUNCH_DARKLY_SDK_KEY: str({
     default: '',
   }),
+  WORKER_DEFAULT_CONCURRENCY: num({
+    default: undefined,
```
Comment (scopsy) [inline apps/api/src/config/env-validator.ts:76]: Should we default to 200 here?

File: packages/application-generic/src/config/workers.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  concurrency: number;
+  lockDuration: number;
+}
+
+type WorkersConfig = Record<WorkerEnum, IWorkerConfig>;
+
+const workersConfig: WorkersConfig = {
+  [WorkerEnum.INBOUND_PARSE_MAIL]: {
+    concurrency: defaultConcurrency ?? 200,
+    lockDuration: defaultLockDuration ?? 900,
```
Comment (scopsy) [inline packages/application-generic/src/config/workers.ts:26]:
```suggestion
    lockDuration: defaultLockDuration ?? 90000,
```

I think the old value is `90000` ?

Comment (p-fernandez) [reply packages/application-generic/src/config/workers.ts:26]: Good catch. I got confused with Active and Completed metrics workers.

**Review by Cliftonz — APPROVED** [review]
lgtm

**Pull request merged**